.github/workflows/community-label.yml

```diff
@@ -0,0 +1,112 @@
+name: Add Community Label
+
+on:
+  issues:
+    types: [opened]
+  pull_request_target:
+    types: [opened]
+
+concurrency: 
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  verify:
+    name: Verify
+    runs-on: ubuntu-20.04
+    outputs:
+      issueId: ${{ steps.check.outputs.issueId }}
+      run: ${{ steps.check.outputs.run }}
+    steps:
+      - name: Check if user is a community contributor
+        id: check
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const types = [ 'issue', 'pull_request' ];
+            const orgs = [ 'automattic', 'woocommerce' ];
+
+            let found = null; // null indicates we haven't done any checks.
+            let type, org, result;
+
+            for ( var typeIndex in types ) {
+              type = types[ typeIndex ];
+
+              if ( ! context.payload[ type ] || ! context.payload[ type ].user || ! context.payload[ type ].user.login ) {
+                // If we don't have the context for this type, move on to checking the next type.
+                console.log( 'No payload found for %s, continuing...', type );
+                continue;
+              }
+
+              if ( context.payload[ type ].user.type && context.payload[ type ].user.type === 'Bot' ) {
+                // Treat bots as non-community contributors.
+                console.log( 'User %s is a bot user. Skipping...', payload[ type ].user.login );
+                found = true;
```

Comment:
This is not really the idiomatic way to do this kind of work in JavaScript. I would avoid mutating a variable in a loop generally and I would also avoid lots of break; and continue;

I can certainly see there is some complexity here though, you're nesting loops and using async code.

I think I would keep the `let found` in this case and break out the async/sync code into functions something like:

```
async function userIsInAnOrgWeCareAbout(username) {
     for (const org of orgs) {
      ....
     } 
}

function userIsBot(contextPayload, type) {
}

function validPayload(contextPayload, type) {
  // If we don't have the context for this type, move on to checking the next type.
}
```

And then the code should be something like:

```
let found = false;
for (const type of types) {
  // rather than using break and continue just don't do anything if payload is not there
  if(validPayload(contextPayload)) {
    if (userIsBot(contextPayload, type)) {
      found = true;
    } else {
       found = await userIsInOrgWeCareAbout(context.payload[ type ].user.login);
    }
  }
}
```